web/modules/custom/solution/src/Form/ChangeCollectionForm.php

```diff
@@ -0,0 +1,221 @@
+    $selection_settings = $selection->getConfiguration() + [
+      'match_operator' => 'CONTAINS',
+    ];
+
+    $form['destination_collection'] = [
+      '#title' => $this->t('Select the destination collection'),
+      '#type' => 'entity_autocomplete',
+      '#target_type' => 'rdf_entity',
+      '#selection_handler' => 'default:rdf_entity',
+      '#selection_settings' => $selection_settings,
+      '#maxlength' => 2048,
+      '#required' => TRUE,
+    ];
+
+    $form['actions'] = [
+      '#type' => 'actions',
+      'submit' => [
+        '#type' => 'submit',
+        '#value' => $this->t('Move solutions'),
+      ],
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state): void {
+    parent::validateForm($form, $form_state);
+    if ($form_state->getValue('source_collection')->id() === $form_state->getValue('destination_collection')) {
+      $form_state->setErrorByName('destination_collection', $this->t('The destination collection cannot be the same as the source collection.'));
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state): void {
+    /** @var \Drupal\rdf_entity\RdfInterface $source_collection */
+    $source_collection = $form_state->getValue('source_collection');
+    $destination_collection_id = $form_state->getValue('destination_collection');
+
+    if (!$destination_collection = $this->getRdfEntityStorage()->load($destination_collection_id)) {
+      throw new \RuntimeException("Cannot load RDF entity with ID $destination_collection_id");
+    }
+
+    foreach ($this->getSolutions() as $solution) {
+      $messenger_arguments = [
+        '%solution' => $solution->toLink()->toString(),
+        '%destination_collection' => $destination_collection->toLink()->toString(),
+      ];
+      $logger_arguments = [
+        '@solution' => $solution->label(),
+        '@destination_collection' => $destination_collection->label(),
+        '@source_collection' => $source_collection->label(),
+      ];
+
+      try {
+        // Prevent notification dispatching.
+        // @see joinup_notification_dispatch_notification()
+        $solution->skip_notification = TRUE;
+        $solution->set('collection', $destination_collection_id)->save();
+        $this->messenger()->addStatus($this->t('Solution %solution has been moved to %destination_collection.', $messenger_arguments));
+        $this->logger->info("Solution '@solution' moved from '@source_collection' to '@destination_collection' collection.", $logger_arguments);
+      }
+      catch (\Exception $e) {
+        $this->messenger()->addError($this->t("Error while trying to change the collection for %solution solution.", $messenger_arguments));
+        $this->logger->info("Error while trying to move solution '@solution' from '@source_collection' to '@destination_collection' collection.", $logger_arguments);
+      }
```

Comment:
In this case I think it would be better to call `$this->logger->error()` rather than `->info()` so that any error that occurs which be easier to find in the logs.